### PR TITLE
fix: remove duplicate item from Newest List

### LINKED ISSUE #3059: Duplicated nft in Newest List
In landing page we have duplicated nfts (Newest List)

Same Id `/rmrk/gallery/12782302-8000262b13545ea72d-KBUTTS-JESSIE_EDITION_KSM_BUTT-0000000000000061`

Capture d’écran 2022-05-23 à 5 06 44 PM

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #3059 
- [x] related with https://github.com/kodadot/rubick/pull/89

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
before:   
image

after fix:    
image

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
IMO it would be better to do something at the query level maybe try to hack with filters

This might introduce edge case where an user LIST more than 10 times and it's going to show only one nft in NewestList

**Comment by Jarsen136** [comment]
> IMO it would be better to do something at the query level maybe try to hack with filters
> 
> This might introduce edge case where an user LIST more than 10 times and it's going to show only one nft in NewestList

Yes, but I think the best solution is to make a resolver for LIST event. ref: https://github.com/kodadot/rubick/issues/29 cc @vikiival 

I will try to make it.

**Comment by Jarsen136** [comment]
> Yes, but I think the best solution is to make a resolver for LIST event. ref: [kodadot/rubick#29](https://github.com/kodadot/rubick/issues/29) cc @vikiival
> 
> I will try to make it.

ref https://github.com/kodadot/rubick/pull/89

**Review by roiLeo — COMMENTED** [review]
With https://github.com/kodadot/rubick/pull/89 merged, will you change fix #3059 or do you have to create a new query?

**Comment by Jarsen136** [comment]
> With [kodadot/rubick#89](https://github.com/kodadot/rubick/pull/89) merged, does it fix #3059 or do you have to create a new query?

Yes, I need to create a new query to apply the new resolver.

**Comment by Jarsen136** [comment]
I just found a bug from the new resolver : )  https://github.com/kodadot/rubick/pull/100

**Comment by yangwao** [comment]
so it's ready to test/merge? :p

**Comment by Jarsen136** [comment]
> so it's ready to test/merge? :p

not yet. I am waiting for its deployment again :)

**Comment by vikiival** [comment]
> I am waiting for its deployment again :)

deploying

**Comment by Jarsen136** [comment]
- [x] remove duplicate item from [Newest List]/[Latest sales]

I found two new bugs : )

- [ ]  inaccurate time about listing

image

image

- [ ] nft may be displayed on [Newest List] even if  its listing price equal to 0.

File: queries/rmrk/subsquid/lastNftListByEvent.graphql [hunk 1]
```diff
@@ -1,21 +1,16 @@
-query lastNftListByEvent($limit: Int!, $event: Interaction!, $and: [EventWhereInput!]) {
-  events(
+query lastNftListByEvent($limit: Float!, $event: String!) {
+  lastEvent(
```
Comment (vikiival) [inline queries/rmrk/subsquid/lastNftListByEvent.graphql:2]:
```suggestion
  events: lastEvent(
```

File: utils/carousel.ts [hunk 2]
```diff
@@ -55,15 +55,16 @@ export const fallbackMetaByNftEvent = async (events: Events[]) => {
...
-  const result: any[] = []
-  list.forEach((item) => {
-    const itemId = item[id]
-    if (!idSet.has(itemId)) {
-      idSet.add(itemId)
-      result.push(item)
-    }
-  })
-  return result
+export const convertLastEventToNft = (e: any) => {
```
Comment (vikiival) [inline utils/carousel.ts:58]: lol `any` :)

**Review by vikiival — CHANGES_REQUESTED** [review]
Smol stuff 

+ please do not copy paste stuff 3 times :)

https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

**Comment by vikiival** [comment]
> nft may be displayed on [Newest List] even if its listing price equal to 0.

this was already solved couple of PRs ago

**Comment by Jarsen136** [comment]
Updated: 
- [x] remove duplicate item from [Newest List]/[Latest sales]

- [x]  inaccurate time to nft listing
https://github.com/kodadot/rubick/pull/101
- [x] nft may be displayed on [Newest List] even if  its listing price equal to 0.

- [ ] support passion feed

**Review by roiLeo — APPROVED** [review]
✅ code lgtm
didn't test yet

**Comment by prachi00** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 55.34 USD/KSM ~ 1.807 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xe409d5d52188536e415e2a66bcbf54d22271f2d2c6d40050e395bce3ba8dee23](https://kusama.subscan.io/extrinsic/0xe409d5d52188536e415e2a66bcbf54d22271f2d2c6d40050e395bce3ba8dee23)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**